### PR TITLE
Add information about mria

### PR DESCRIPTION
**Pull request by ieQu1**
[ ] Check CN translation

### CODE REVIEW

File: en_US/deploy/cluster/db.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+## Monitoring and troubleshooting
+
+Monitoring of the Mria performance can be done using Prometheus metrics or using Erlang console.
+
+### Prometheus metrics
+#### Core
+- `emqx_mria_last_intercepted_trans`: Number of transactions received by the shard since the node start. Note that this value can be different on different core nodes.
+- `emqx_mria_weight`: A value used for load balancing. It changes depending on the momentary load of the core node.
+- `emqx_mria_replicants`: Number of replicants connected to the core node, replicating data for the given shard.
+- `emqx_mria_server_mql`: Number of unprocessed transactions, waiting to be sent to the replicants. Less is better. If this metric tends to grow, more core nodes is needed.
```
Comment (thalesmg) [inline en_US/deploy/cluster/db.md]:
```suggestion
- `emqx_mria_server_mql`: Number of unprocessed transactions, waiting to be sent to the replicants. Less is better. If this metric tends to grow, more core nodes are needed.
```

File: en_US/deploy/cluster/db.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+#### Replicant
+- `emqx_mria_lag`: Replicant lag, indicating how far behind the upstream core node the replicant lags. Less is better.
+- `emqx_mria_bootstrap_time`: Time spent during bootstrapping of the replicant. This value doesn't change during normal operation of the replicant
+- `emqx_mria_bootstrap_num_keys`: Number of database records copied from the core node during bootstrap. This value doesn't change during normal operation of the replicant
+- `emqx_mria_message_queue_len`: Message queue length of the replica process. Should be around 0 all the time.
+- `emqx_mria_replayq_len`: Length of the internal replay queue on the replicant. Less is better.
+
+### Console commands
+
+`mria_rlog:status().` command can be executed in the EMQX erlang shell to get more extensive information about the state of the embedded database.
```
Comment (thalesmg) [inline en_US/deploy/cluster/db.md]:
```suggestion
`mria_rlog:status().` command can be executed in the EMQX Erlang shell to get more extensive information about the state of the embedded database.
```

**Review by thalesmg — DISMISSED** [review]

File: en_US/deploy/cluster/db.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+If the size of the EMQX cluster is below 5 nodes, typically no tuning of the embedded database is required.
+
+However for horizontal scalability it is recommended to split the nodes in the cluster into two groups:
+
+- core nodes
+- replicant nodes
+
+## Node roles
+
+*Core nodes* serve as a source of truth for the database: they are connected in a full mesh, and each one of them contains an up-to-date replica of the data.
```
Comment (thalesmg) [inline en_US/deploy/cluster/db.md:23]: Maybe we shouldn't mention that the replicants also are in a full mesh?  Just not a Mnesia consensus one?

Comment (ieQu1) [reply en_US/deploy/cluster/db.md:23]: In this context I think it will be quite confusing.

**Review by thalesmg — DISMISSED** [review]

File: en_US/deploy/cluster/db.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+- configuration
+- alarms
+- etc.
+
+Mria tables are replicated across all EMQX nodes.
+It also helps with the fault-tolerance: the data is safe, as long as at least one node in the cluster is alive.
+
+If the size of the EMQX cluster is below 5 nodes, typically no tuning of the embedded database is required.
+
+However for horizontal scalability it is recommended to split the nodes in the cluster into two groups:
```
Comment (qzhuyan) [inline en_US/deploy/cluster/db.md]: However,

**Review by thalesmg — DISMISSED** [review]

File: zh_CN/deploy/cluster/db.md [hunk 5]
```diff
@@ -33,20 +33,20 @@ Mria表在所有EMQX节点上进行复制。
...
+- 它可以实现复制者集群的自动伸缩
 
 由于复制者节点不参与写入，当更多的复制者加入集群时，表的更新效率不会受到影响。
 这允许创建更大的EMQX集群。
 
 另外，复制体节点被设计成短暂的。
-添加或删除它们不会改变数据冗余，所以它们可以被放在一个自动扩展组中，从而实现更好的DevOps实践。
-请注意，根据数据的大小，从核心节点初始复制数据是一个相对繁重的操作，所以自动缩放策略不能非常激进。
+添加或删除它们不会改变数据冗余，所以它们可以被放在一个自动伸缩组中，从而实现更好的DevOps实践。
+请注意，根据数据的大小，从核心节点初始复制数据是一个相对繁重的操作，所以自动缩放策略不能过于激进。
```
Comment (qzhuyan) [inline zh_CN/deploy/cluster/db.md]: 自动伸缩

File: zh_CN/deploy/cluster/db.md [hunk 6]
```diff
@@ -1,30 +1,30 @@
...
+EMQX使用一个嵌入式数据库[Mria](https://github.com/emqx/mria)来存储以下信息。
 
 - 路由表
 - 会话
 - 配置
 - 警报
 - 等。
 
-Mria表在所有EMQX节点上进行复制。
+Mria的表会被复制到所有所有EMQX节点上。
```
Comment (qzhuyan) [inline zh_CN/deploy/cluster/db.md]: 所有

File: zh_CN/deploy/cluster/db.md [hunk 7]
```diff
@@ -33,20 +33,20 @@ Mria表在所有EMQX节点上进行复制。
 
 - 横向可扩展性（我们已经测试了有23个节点的EMQX集群）
 
-- 它可以实现复制者集群的自动缩放
+- 它可以实现复制者集群的自动伸缩
```
Comment (qzhuyan) [inline zh_CN/deploy/cluster/db.md]: 复制节点

File: en_US/deploy/cluster/auto.md [hunk 8]
```diff
@@ -69,7 +78,7 @@ Automatically discover and create clusters based on [Kubernetes](https://kuberne
 ```bash
 cluster.discovery = k8s
 cluster.k8s.apiserver = http://10.110.111.204:8080
-cluster.k8s.service_name = ekka
+cluster.k8s.service_name = emqx
 cluster.k8s.address_type = ip
-cluster.k8s.app_name = ekka
+cluster.k8s.app_name = emqx
```
Comment (zmstone) [inline en_US/deploy/cluster/auto.md]: this is removed.

File: en_US/deploy/cluster/db.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
+# Embedded database
```
Comment (zmstone) [inline en_US/deploy/cluster/db.md]:
```suggestion
# Cluster Scalability
```

File: en_US/deploy/cluster/db.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+Adding or removing them won't change the data redundancy, so they can be placed in an autoscaling group, thus enabling better DevOps practices.
+Note, that initial replication of the data from the core nodes is a relatively heavy operation, depending on the size of the data, so autoscaling policy must not be very aggressive.
+
+## Configuration
+
+In EMQX 5.0 all nodes assume core role by default, so without any tweaks the cluster behaves like 4.*.
+
+To make use of the new replication protocol, set `EMQX_NODE__DB_ROLE` environment variable or `node.db_role` setting in `emqx.conf` to `replicant` on some of the nodes in the cluster, this way they will assume replicant role. Note that there must be at least one core node in the cluster, we recommend 3 cores + N replicants setup as the starting point.
+
+Core nodes may accept MQTT traffic, or they can serve purely as the database server for the replicants, depending on the usecase.
```
Comment (zmstone) [inline en_US/deploy/cluster/db.md]:
```suggestion
Core nodes may accept MQTT traffic, or they can disable all MQTT listeners to serve purely as the database server for the replicants, depending on the usecase.
```

File: en_US/deploy/cluster/db.md [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+#### Replicant
+- `emqx_mria_lag`: Replicant lag, indicating how far behind the upstream core node the replicant lags. Less is better.
+- `emqx_mria_bootstrap_time`: Time spent during bootstrapping of the replicant. This value doesn't change during normal operation of the replicant
+- `emqx_mria_bootstrap_num_keys`: Number of database records copied from the core node during bootstrap. This value doesn't change during normal operation of the replicant
+- `emqx_mria_message_queue_len`: Message queue length of the replica process. Should be around 0 all the time.
+- `emqx_mria_replayq_len`: Length of the internal replay queue on the replicant. Less is better.
+
+### Console commands
+
+`mria_rlog:status().` command can be executed in the EMQX Erlang shell to get more extensive information about the state of the embedded database.
```
Comment (zmstone) [inline en_US/deploy/cluster/db.md]:
```suggestion
`emqx eval 'mria_rlog:status().'` command can be executed to get more extensive information about the state of the embedded database.
```

File: zh_CN/deploy/cluster/db.md [hunk 12]
```diff
@@ -0,0 +1,77 @@
+# 嵌入式数据库
+
+EMQX使用一个嵌入式数据库[Mria](https://github.com/emqx/mria)来存储以下信息。
+
+- 路由表
+- 会话
+- 配置
+- 警报
+- 等。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- 等等
```

File: en_US/deploy/cluster/db.md [hunk 13]
```diff
@@ -0,0 +1,77 @@
...
+- routing table
+- sessions
+- configuration
+- alarms
+- etc.
+
+Mria tables are replicated across all EMQX nodes.
+It also helps with the fault-tolerance: the data is safe, as long as at least one node in the cluster is alive.
+
+If the size of the EMQX cluster is below 5 nodes, typically no tuning of the embedded database is required.
```
Comment (zmstone) [inline en_US/deploy/cluster/db.md]:
```suggestion
If the size of the EMQX cluster is below 5 nodes, typically no database scalability tuning is required.
```

File: zh_CN/deploy/cluster/db.md [hunk 14]
```diff
@@ -0,0 +1,77 @@
...
+
+Mria的表会被复制到所有EMQX节点上。
+这也有助于容错：只要集群中至少有一个节点是活的，数据就是安全的。
+
+如果EMQX集群的规模低于5个节点，通常不需要对嵌入式数据库进行调整。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
如果EMQX集群的规模低于5个节点，通常不需要对数据库的弹性伸缩能力进行调优。
```

File: zh_CN/deploy/cluster/db.md [hunk 15]
```diff
@@ -0,0 +1,77 @@
...
+
+然而，为了实现横向扩展，建议将集群中的节点分成两类（角色）。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
然而，为了实现横向扩展，建议将集群中的节点分成两个角色。
```

File: zh_CN/deploy/cluster/db.md [hunk 16]
```diff
@@ -0,0 +1,77 @@
...
+
+- 核心节点
+- 复制者节点
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- 核心（Core）节点 
- 复制（Replicant）节点
```

File: zh_CN/deploy/cluster/db.md [hunk 17]
```diff
@@ -0,0 +1,77 @@
...
+
+## 节点角色
+
+*核心节点*作为数据库的真数据源：它们以全网状连接，每一个节点都包含一个最新的数据副本。
+核心节点一般是静态和持久的，也就是说，不建议对核心集群进行自动伸缩。
+
+另一方面，*复制者节点*，将所有表的写操作转到核心节点上处理。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
另一方面，*复制节点*，将所有表的写操作转到核心节点上处理。
```

File: zh_CN/deploy/cluster/db.md [hunk 18]
```diff
@@ -0,0 +1,77 @@
...
+它们连接到其中一个核心节点，并被动地复制核心节点的事务。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
它们各自连接到核心节点的其中一个，并被动地复制核心节点的事务。
```

File: zh_CN/deploy/cluster/db.md [hunk 19]
```diff
@@ -0,0 +1,77 @@
...
+这意味着复制者节点不允许自己执行任何写操作。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
这意味着复制节点不允许自己执行任何写操作。
```

File: zh_CN/deploy/cluster/db.md [hunk 20]
```diff
@@ -0,0 +1,77 @@
...
+相反，它们要求核心节点代表它们更新数据。
+同时，他们有一个完整的本地数据副本，所以读取访问的速度也同样快。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
同时，它们有一个完整的本地数据副本，所以读取访问的速度也同样快。
```

File: zh_CN/deploy/cluster/db.md [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+
+这种方法解决了两个问题。
+
+- 横向可扩展性（我们已经测试了有23个节点的EMQX集群）
+
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md:36]:
```suggestion
```

File: zh_CN/deploy/cluster/db.md [hunk 22]
```diff
@@ -0,0 +1,77 @@
...
+- 它可以实现复制者节点集群的自动伸缩
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- 它可以实现复制节点集群的自动伸缩
```

File: zh_CN/deploy/cluster/db.md [hunk 23]
```diff
@@ -0,0 +1,77 @@
...
+
+由于复制者节点不参与写入，当更多的复制者加入集群时，表的更新效率不会受到影响。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
由于复制节点不参与数据写入，当更多的复制者加入集群时，表的更新效率不会受到影响。
```

File: zh_CN/deploy/cluster/db.md [hunk 24]
```diff
@@ -0,0 +1,77 @@
...
+这允许创建更大的EMQX集群。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
进而允许创建更大的EMQX集群。
```

File: zh_CN/deploy/cluster/db.md [hunk 25]
```diff
@@ -0,0 +1,77 @@
...
+
+另外，复制体节点被设计成短暂的。
+添加或删除它们不会改变数据冗余，所以它们可以被放在一个自动伸缩组中，从而实现更好的DevOps实践。
+请注意，根据数据的大小，从核心节点初始复制数据是一个相对繁重的操作，所以自动伸缩策略不能过于激进。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
另外，复制节点被设计成可以按需增删，
添加或删除它们不会改变数据冗余，所以它们可以被放在一个自动伸缩组中，从而实现更好的DevOps实践。
请注意，根据总数据量的大小，从核心节点初始复制数据是一个相对繁重的操作，所以自动伸缩策略不能过于激进。
```

File: zh_CN/deploy/cluster/db.md [hunk 26]
```diff
@@ -0,0 +1,77 @@
...
+
+## 配置
+
+在EMQX 5.0中，所有节点都默认为核心角色，所以不做任何调整，集群的行为就像4.*。
+
+为了使用新的复制协议，在集群中的一些节点上设置`EMQX_NODE__DB_ROLE`环境变量或`emqx.conf`中的`node.db_role`设置为`replicant`，这样它们将承担复制者角色。请注意，集群中必须至少有一个核心节点，我们建议以3个核心+N个复制者的设置为最初配置。
+
+核心节点可以接受MQTT流量，或者它们可以纯粹作为复制者的数据库服务器，这取决于使用情况。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
根据业务需求，核心节点即可以接受MQTT流量，也可以把所有的MQTT监听器关闭，从而变成纯粹的数据库服务器。
```

File: zh_CN/deploy/cluster/db.md [hunk 27]
```diff
@@ -0,0 +1,77 @@
...
+
+- 在一个小集群中（总共3个节点或更少），使用复制体并不经济，所以核心节点承担所有的流量。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- 在一个小集群中（总共3个节点或更少），引入复制节点并不经济，所以可以由核心节点承担所有的流量。
```

File: zh_CN/deploy/cluster/db.md [hunk 28]
```diff
@@ -0,0 +1,77 @@
...
+- 在一个非常大的集群中（10个节点或更多），将流量从核心节点移开是有意义的。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- 在一个非常大的集群中（例如10或更多个节点），核心+复制节点集训的优势可以很好的发挥出来。
```

File: zh_CN/deploy/cluster/db.md [hunk 29]
```diff
@@ -0,0 +1,77 @@
...
+- 在一个中等规模的集群中，这真的取决于许多因素，所以需要进行实验。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- 在一个中等规模的集群中，节点数量的选取取决于诸多因素，所以根据模拟生产环境的场景测试情况进行灵活选择。
```

File: zh_CN/deploy/cluster/db.md [hunk 30]
```diff
@@ -0,0 +1,77 @@
...
+
+## 监测和故障排除
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
## 监控和故障排查
```

File: en_US/deploy/cluster/db.md [hunk 31]
```diff
@@ -0,0 +1,77 @@
...
+
+## Monitoring and troubleshooting
+
+Monitoring of the Mria performance can be done using Prometheus metrics or using Erlang console.
+
+### Prometheus metrics
+#### Core
+- `emqx_mria_last_intercepted_trans`: Number of transactions received by the shard since the node start. Note that this value can be different on different core nodes.
+- `emqx_mria_weight`: A value used for load balancing. It changes depending on the momentary load of the core node.
+- `emqx_mria_replicants`: Number of replicants connected to the core node, replicating data for the given shard.
```
Comment (zmstone) [inline en_US/deploy/cluster/db.md]: ?
```suggestion
- `emqx_mria_replicants`: Number of replicants connected to the core node. Numbers are grouped per shard.
```

File: zh_CN/deploy/cluster/db.md [hunk 32]
```diff
@@ -0,0 +1,77 @@
...
+
+对Mria性能的监控可以通过Prometheus指标或使用Erlang控制台来完成。
+
+### Prometheus 系统指标
+#### 核心节点
+- `emqx_mria_last_intercepted_trans`: 自节点启动以来，分片区收到的事务数量。注意，这个值在不同的核心节点上可能不同。
+- `emqx_mria_weight`: 一个用于负载平衡的值。它的变化取决于核心节点的瞬间负载。
+- `emqx_mria_replicants`: 连接到核心节点的复制者节点的数量，复制给定分片的数据。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- `emqx_mria_replicants`: 连接到核心节点的复制节点数量。
```
not sure what the second part means, so removed it for now.

File: zh_CN/deploy/cluster/db.md [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+- `emqx_mria_server_mql`: 未处理的事务数量，等待发送至复制者节点。越少越好。如果这个指标有增长的趋势，需要更多的核心节点。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- `emqx_mria_server_mql`: 未处理的事务数量，等待发送至复制者节点。越少越好。如果这个指标有增长的趋势，则可能需要为当前的核心节点增加算力资源，或添加更多的核心节点。
```

File: en_US/deploy/cluster/db.md [hunk 34]
```diff
@@ -0,0 +1,77 @@
...
+- `emqx_mria_server_mql`: Number of unprocessed transactions, waiting to be sent to the replicants. Less is better. If this metric tends to grow, more core nodes are needed.
```
Comment (zmstone) [inline en_US/deploy/cluster/db.md]:
```suggestion
- `emqx_mria_server_mql`: Number of unprocessed transactions, waiting to be sent to the replicants. Less is better. If this metric tends to grow, then it's probably time to add more computing resource for the exiting core nodes, or even add more core nodes.
```

File: zh_CN/deploy/cluster/db.md [hunk 35]
```diff
@@ -0,0 +1,77 @@
...
+
+#### 复制者节点
+- `emqx_mria_lag`。复制者节点滞后，表示复制者节点滞后上游核心节点的程度。越少越好。
+- `emqx_mria_bootstrap_time`: 复制者节点启动过程中花费的时间。这个值在复制者节点的正常运行过程中不会改变。
+- `emqx_mria_bootstrap_num_keys`: 在引导过程中从核心节点复制的数据库记录的数量。这个值在复制者节点的正常运行中不会改变。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
- `emqx_mria_bootstrap_num_keys`: 在初始复制过程中从核心节点复制的数据库记录的数量。这个值在复制者节点的正常运行中不会改变。
```

File: zh_CN/deploy/cluster/db.md [hunk 36]
```diff
@@ -0,0 +1,77 @@
...
+- `emqx_mria_message_queue_len`: 复制进程的消息队列长度。应该一直保持在0左右。
+- `emqx_mria_replayq_len`: 复制者节点的内部重放队列的长度。越少越好。
+
+### 控制台命令
+
+`mria_rlog:status().`此命令可以在EMQX erlang shell中执行，以获得关于嵌入式数据库状态的更多信息。
```
Comment (zmstone) [inline zh_CN/deploy/cluster/db.md]:
```suggestion
执行 `emqx_ctl eval 'mria_rlog:status().'` 命令可以获得关于嵌入式数据库状态的更多信息。
```

**Review by zmstone — APPROVED** [review]

**Pull request merged**